internal/exec/engine.go

```diff
@@ -274,8 +276,21 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.ConfigReference) (types.Conf
 	if err != nil {
 		return types.Config{}, err
 	}
+	headers := http.Header{}
+	for headerName, headerValue := range resource.ConfigHeaders {
+		headers.Add(headerName, headerValue[0])
+	}
+	if cfgRef.HTTPHeaders != nil && len(cfgRef.HTTPHeaders) > 0 {
+		if u.Scheme != "http" && u.Scheme != "https" {
+			return types.Config{}, e.Logger.Crit("Cannot use HTTP headers with %v source scheme", u.Scheme)
+		}
+		for _, headerData := range cfgRef.HTTPHeaders {
+			header := strings.Split(string(headerData), ":")
```

Comment:
The value part of the header might contain a `:`, therefore you might want to use something like `strings.SplitN(string(headerData), ":", 2)`